src/material-examples/table-overview/table-overview-example.ts

```diff
@@ -1,69 +1,60 @@
-import {Component, ElementRef, ViewChild} from '@angular/core';
-import {DataSource} from '@angular/cdk/collections';
-import {MatPaginator, MatSort} from '@angular/material';
-import {SelectionModel} from '@angular/cdk/collections';
-import {BehaviorSubject} from 'rxjs/BehaviorSubject';
-import {Observable} from 'rxjs/Observable';
-import 'rxjs/add/operator/startWith';
-import 'rxjs/add/observable/merge';
-import 'rxjs/add/observable/fromEvent';
-import 'rxjs/add/operator/map';
-import 'rxjs/add/operator/distinctUntilChanged';
-import 'rxjs/add/operator/debounceTime';
+import {Component, ViewChild} from '@angular/core';
+import {MatPaginator, MatSort, MatTableDataSource} from '@angular/material';
 
 /**
- * @title Feature-rich data table
+ * @title Data table with sorting, pagination, and filtering.
  */
 @Component({
   selector: 'table-overview-example',
   styleUrls: ['table-overview-example.css'],
   templateUrl: 'table-overview-example.html',
 })
 export class TableOverviewExample {
-  displayedColumns = ['select', 'userId', 'userName', 'progress', 'color'];
-  exampleDatabase = new ExampleDatabase();
-  selection = new SelectionModel<string>(true, []);
-  dataSource: ExampleDataSource | null;
+  displayedColumns = ['id', 'name', 'progress', 'color'];
+  dataSource: MatTableDataSource<UserData>;
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
-  @ViewChild('filter') filter: ElementRef;
 
-  ngOnInit() {
-    this.dataSource = new ExampleDataSource(this.exampleDatabase, this.paginator, this.sort);
-    Observable.fromEvent(this.filter.nativeElement, 'keyup')
-        .debounceTime(150)
-        .distinctUntilChanged()
-        .subscribe(() => {
-          if (!this.dataSource) { return; }
-          this.dataSource.filter = this.filter.nativeElement.value;
-        });
-  }
-
-  isAllSelected(): boolean {
-    if (!this.dataSource) { return false; }
-    if (this.selection.isEmpty()) { return false; }
+  constructor() {
+    // Create 100 users
+    const users: UserData[] = [];
+    for (let i = 1; i <= 100; i++) { users.push(createNewUser(i)); }
 
-    if (this.filter.nativeElement.value) {
-      return this.selection.selected.length == this.dataSource.renderedData.length;
-    } else {
-      return this.selection.selected.length == this.exampleDatabase.data.length;
-    }
+    // Assign the data to the data source for the table to render
+    this.dataSource = new MatTableDataSource(users);
   }
 
-  masterToggle() {
-    if (!this.dataSource) { return; }
+  /**
+   * Set the paginator and sort after the view has been initialized so
+   * that the components are completely set up for the data source to access.
+   */
```

Comment:
Is there any other way to word this? "...so that component are completely set up..." kind of glosses over why it has to be here instead of in, say, OnInit.

Maybe
> Set the paginator and sort after the view has been initialized since that is
the first time the component has access to the ViewChild queries.

Reply:
Changed to 

> Set the sort after the view init since this component will be able to query its view for the initialized sort.

Reply:
> Wait until after the view init to set the paginator and sort in the data source. This will enable the component to query its view for the initialized paginator and sort components.

Maybe an improvement? I guess it doesn't really matter, and I think having any comment at all should signify that it's important to keep it there after the copy-paste.